pkg/services/auth/auth_token.go

```diff
@@ -133,6 +141,10 @@ func (s *UserAuthTokenService) LookupToken(ctx context.Context, unhashedToken st
 		return nil, models.ErrUserTokenNotFound
 	}
 
+	if model.CreatedAt <= s.createdAfterParam() || model.RotatedAt <= s.rotatedAfterParam() {
+		return &models.UserToken{Id: model.Id, UserId: model.UserId}, models.ErrUserTokenExpired
```

Comment:
Feels weird to return both a token and an error. Have you considered handling this scenario in a background job instead? I'm not sure that makes things easier though.

Reply:
I created a new struct `TokenExpiredError` to hold more information in the error itself and not return the token anymore, WDYT?